src/test/java/uk/gov/companieshouse/bankruptofficersearch/api/service/ScottishBankruptOfficerSearchServiceImplTest.java

```diff
@@ -0,0 +1,117 @@
+import uk.gov.companieshouse.bankruptofficersearch.api.model.response.ScottishBankruptOfficerSearchResultsEntity;
+import uk.gov.companieshouse.bankruptofficersearch.api.model.rest.ScottishBankruptOfficerDetails;
+import uk.gov.companieshouse.bankruptofficersearch.api.model.rest.ScottishBankruptOfficerSearch;
+import uk.gov.companieshouse.bankruptofficersearch.api.model.rest.ScottishBankruptOfficerSearchFilters;
+import uk.gov.companieshouse.bankruptofficersearch.api.model.rest.ScottishBankruptOfficerSearchResults;
+import uk.gov.companieshouse.bankruptofficersearch.api.request.OracleQueryDaoImpl;
+import uk.gov.companieshouse.bankruptofficersearch.api.service.impl.ScottishBankruptOfficerSearchServiceImpl;
+import uk.gov.companieshouse.bankruptofficersearch.api.transformer.ScottishBankruptOfficerTransformer;
+
+@ExtendWith(MockitoExtension.class)  class ScottishBankruptOfficerSearchServiceImplTest {
+
+    @Mock
+    private OracleQueryDaoImpl dao;
+
+    @Mock
+    private ScottishBankruptOfficerTransformer transformer;
+
+    @InjectMocks
+    private ScottishBankruptOfficerSearchServiceImpl service;
+
+    private static final int START_INDEX = 1;
+    private static final int ITEMS_PER_PAGE = 2;
+    private static final String FORENAME = "forename";
+    private static final String SURNAME = "surname";
+    private static final String DATE_OF_BIRTH = "2020-01-01";
+    private static final String POSTCODE = "postcode";
+    private static final String EPHEMERAL_KEY = "0123456";
+
+
+    @Test
+    @DisplayName("Search for bankrupt officers")
+    void testScottishBankruptSearch() {
+        ScottishBankruptOfficerSearchFilters searchFilters = new ScottishBankruptOfficerSearchFilters();
+        searchFilters.setForename1(FORENAME);
+        searchFilters.setSurname(SURNAME);
+        searchFilters.setDateOfBirth(DATE_OF_BIRTH);
+        searchFilters.setPostcode(POSTCODE);
+
+        ScottishBankruptOfficerSearch search = new ScottishBankruptOfficerSearch();
+        search.setStartIndex(START_INDEX);
+        search.setItemsPerPage(ITEMS_PER_PAGE);
+        search.setFilters(searchFilters);
+
+        ScottishBankruptOfficerSearchEntity searchEntity = new ScottishBankruptOfficerSearchEntity();
+        ScottishBankruptOfficerSearchResultsEntity resultsEntity = new ScottishBankruptOfficerSearchResultsEntity();
+        ScottishBankruptOfficerSearchResults expectedSearchResults = new ScottishBankruptOfficerSearchResults();
+
+        when(transformer.convertToSearchEntity(search)).thenReturn(searchEntity);
+        when(dao.getScottishBankruptOfficers(searchEntity)).thenReturn(resultsEntity);
+        when(transformer.convertToSearchResults(resultsEntity)).thenReturn(expectedSearchResults);
+
+        ScottishBankruptOfficerSearchResults searchResults = service.searchScottishBankruptOfficers(search);
+        assertEquals(searchResults, expectedSearchResults);
```

Comment:
This should be `assertEquals(expectedSearchResults, searchResults);`

Reply:
Changed